packages/components/src/date-time/time.js

```diff
@@ -277,6 +279,35 @@ class TimePicker extends Component {
 	render() {
 		const { is12Hour } = this.props;
 		const { year, minutes, hours, am } = this.state;
+
+		const TimeZone = () => {
+			const { timezone } = getDateSettings();
+			const offset =
+				timezone.offset > 0 ? '+' + timezone.offset : timezone.offset;
+			const zoneAbbr = moment
+				.tz( new Date(), timezone.string )
+				.zoneAbbr();
+			const timezoneDetail =
+				'UTC' === timezone.string
+					? __( 'Coordinated Universal Time' )
+					: '(UTC' +
+					  offset +
+					  ') ' +
+					  timezone.string.replace( '_', ' ' );
+			const timezoneAbbr =
+				'' !== timezone.string && isNaN( zoneAbbr )
+					? zoneAbbr
+					: 'UTC' + offset;
+
+			return (
+				<Tooltip position="top center" text={ timezoneDetail }>
+					<div className="components-datetime__timezone">
+						{ timezoneAbbr }
+					</div>
+				</Tooltip>
+			);
+		};
```

Comment:
Unless I've missed something, this looks pretty standalone. Let's pull the declaration out of the `render` function